helpdesk/models.py

```diff
@@ -1012,6 +1004,54 @@ def get_markdown(self):
     def time_spent_formated(self):
         return format_time_spent(self.time_spent)
 
+    def time_spent_calculation(self):
+        "Returns timedelta according to rules settings."
+
+        # extract parameters from previous follow-up or ticket
+        try:
+            prev_fup_qs = self.ticket.followup_set.all()
```

Comment:
Here, we don't want to exclude any follow-up, just exclude time calculation if the previous followup status is considered to be excluded.

Here is a example:
When a ticket is set to a *PAUSE* status, the last follow-up will have the *PAUSE* status.
That ticket is kind of on hold or freezed until a new status is set.
When the new *RESUME* status is set, we want to calculate the time between the previous *PAUSE* and current *RESUME* follow-up.
As PAUSE is a status exclusion for time clculation, we will consider the time from PAUSE to RESUME to be 0.

I hope I made myself clear.

Reply:
Alright thanks, I better understand with that example!

I only see one potential issue, and this is related to the previous comment, suppose you make a follow-up and set new status to PAUSE, then later you make another follow-up without changing status. I believe `new_status` property will be None and that could make next interval to count whereas it shouldn't, since ticket was still on hold.

Reply:
```
>>> [f.new_status for f in t.followup_set.all()]
[None, None, None, None, None, None, 6, None, None, None, None, 7, None, None, None, 3, 2, 6, None]
```